health/tracker.go

```diff
@@ -750,6 +751,23 @@ func (t *Tracker) SetLatestBlockNumber(upstream common.Upstream, blockNumber int
 		g := t.getLatestBlockGauge(t.projectId, "*", netLabel, "*")
 		g.Set(float64(blockNumber))
 		needsGlobalUpdate = true
+
+		// Atomically update timestamp when network-level block number is updated
+		if blockTimestamp > 0 {
+			oldTimestamp := ntwMeta.evmLatestBlockTimestamp.Load()
+			if blockTimestamp > oldTimestamp {
```

Comment:
Why do we have this check?

Reply:
redundant check because of `if blockNumber > oldNtwVal` check above it. fixed